server/src/main.rs

```diff
@@ -691,6 +748,15 @@ fn trace_stream(res: &mut Response<Streaming>, child: &mut child::Thread) -> io:
     let mut terminated = false;
     let mut done = false;
     while !done {
+        match kill.0.load(Ordering::Relaxed) {
+            true => {
+                println!("Sending stop from main");
+                child.tx.send(ServerMessage::Stop).unwrap();   
+            },
+            false => {
+                child.tx.send(ServerMessage::Continue).unwrap();   
+            }
+        };
```

Comment:
Shouldn't need this here, since the cancel endpoint should just signal the child thread directly. This function would just receive a `DebugTrace::Cancel` (currently called `StopRequest`).